lighthouse-extension/app/src/popup.js

```diff
@@ -47,72 +48,88 @@ document.addEventListener('DOMContentLoaded', _ => {
-    const attributes = [['type', 'checkbox'], ['value', text]];
+    input.setAttribute('type', 'checkbox');
+    input.setAttribute('value', text);
     if (isChecked) {
-      attributes.push(['checked', 'checked']);
+      input.setAttribute('checked', 'checked');
     }
 
-    attributes.forEach(attr => input.setAttribute.apply(input, attr));
-
     const label = document.createElement('label');
     label.appendChild(input);
     label.appendChild(document.createTextNode(text));
     const listItem = document.createElement('li');
     listItem.appendChild(label);
 
     return listItem;
-  };
-
-  const generateOptionsList = (list, selectedAudits) => {
+  }
+
+  /**
+   * Generates a document fragment containing a list of checkboxes and labels
+   * for the aggregation categories.
+   * @param {!Object<boolean>} selectedAggregations
+   * @return {!DocumentFragment}
+   */
+  function generateOptionsList(list, selectedAggregations) {
     const frag = document.createDocumentFragment();
 
-    aggregations.forEach(aggregation => {
-      frag.appendChild(createOptionItem(aggregation.name, selectedAudits[aggregation.name]));
+    defaultAggregations.forEach(aggregation => {
+      const isChecked = selectedAggregations[aggregation.name];
+      frag.appendChild(createOptionItem(aggregation.name, isChecked));
     });
 
-    list.appendChild(frag);
-  };
+    return frag;
+  }
+
+  /**
+   * Returns an array of names of audits from the selected aggregation
+   * categories.
+   * @param {!Object<boolean>} selectedAggregations
+   * @return {!Array<string>}
+   */
+  function getAuditsFromSelected(selectedAggregations) {
+    const auditLists = defaultAggregations.filter(aggregation => {
+      return selectedAggregations[aggregation.name];
+    }).map(selectedAggregation => {
+      return selectedAggregation.audits;
+    });
 
-  const getAuditsFromCategory = audits => _flatten(
-    Object.keys(audits).filter(audit => audits[audit]).map(audit => {
-      const auditsInCategory = aggregations
-        .find(aggregation => aggregation.name === audit).criteria;
-
-      return Object.keys(auditsInCategory);
-    })
-  );
+    return _flatten(auditLists);
+  }
 
   background.listenForStatus(logstatus);
-  background.fetchAudits().then(audits => {
-    generateOptionsList(optionsList, audits);
+  background.loadSelectedAggregations().then(aggregations => {
+    const frag = generateOptionsList(optionsList, aggregations);
+    optionsList.appendChild(frag);
   });
 
   generateReportEl.addEventListener('click', () => {
     startSpinner();
     feedbackEl.textContent = '';
 
-    background.fetchAudits()
-    .then(getAuditsFromCategory)
-    .then(audits => background.runAudits({
-      flags: {
-        mobile: true,
-        loadPage: true
-      }
-    }, audits))
+    background.loadSelectedAggregations()
+    .then(getAuditsFromSelected)
+    .then(selectedAudits => {
+      return background.runLighthouse({
```

Comment:
just a (debatable :) readability improvement. It's often easy to miss that a multiline body on an arrow function is being returned